index.d.ts

```diff
@@ -0,0 +1,35 @@
+type loggerMethod1 = (x: string | Error) => void
+
+type LaziFy<T> = T | (() => T)
+
+type loggerMethod1Lazy = (
+  lazyX: LaziFy<Parameters<loggerMethod1>[0]>,
+) => void
+
+// TODO: be more specific about y
```

Comment:
`y` can also contain an `error: Error` property which will be treated similar to method1, I wonder if we can capture that in the type

Reply:
Would you be able to propose a more specific type for y?

Reply:
I'm happy to leave it for another task. I'd like to rewrite the logger in typescript (created a task for it), at which point we can define a strict interface.

Reply:
No its not possible to have anything more specific as `y` can be anything. I was only wondering if we can somehow document that a specific property means something, e.g.:

```ts
type y = { error: Error } & Record<string, any>
```

But I think the second part will encompass the more specific error component so it won't be seen in the IDE

Reply:
ok, let's address this in https://darcs.atlassian.net/browse/JUP-428 by making the interface\implementation stricter

Reply:
I expanded the spec for the above ticket with a proposal of strict interface for the logging functions